src/components/ShareMyLocationButton.js

```diff
@@ -28,20 +28,33 @@ const ShareMyLocationButton = ({ disabled = false }) => {
 
   const handleShareLiveLocation = () => {
     if (navigator.geolocation) {
-      if (trackingId === null) {
-        const newTrackingId = navigator.geolocation.watchPosition(
-          handleSuccess,
-          handleError,
-        );
-        setTrackingId(newTrackingId);
-      }
+      const trackingId = navigator.geolocation.watchPosition(
+        handleSuccess,
+        handleError,
+      );
+
+      setTrackingId(trackingId);
+      setUserLocationShared(true);
     }
   };
 
+  const handleStopSharing = () => {
+    if (userLocationShared && trackingId) {
+      setTrackingId(navigator.geolocation.clearWatch(trackingId));
+    }
+    setUserLocationShared(false);
+    setNewCenterMap(defaultCenterMap);
+  };
+  console.log('---', trackingId);
   return (
-    <button onClick={handleShareLiveLocation} disabled={disabled}>
-      Share My Location
-    </button>
+    <>
+      <button onClick={handleShareLiveLocation} disabled={userLocationShared}>
```

Comment:
```suggestion
      <button onClick={userLocationShared ? handleStopSharing : handleShareLiveLocation}>
           {buttonText}
      </button>
```

Reply:
I'd do something like this ^ and just render the exact same button with a different text and different click handler depending on `userLocationShared`.

Then above you'd have to declare:
`const buttonText = userLocationShared ? 'Stop Sharing Location'  : 'Share My Location';`

Reply:
I agree, we mentioned possibly doing this based on the last pr and I went ahead and made that change.